packages/flutter_tools/test/general.shard/commands/build_aar_test.dart

```diff
@@ -90,18 +89,14 @@ void main() {
     Directory tempDir;
     AndroidSdk mockAndroidSdk;
     Usage mockUsage;
-    FileSystem memoryFileSystem;
 
     setUp(() {
       mockUsage = MockUsage();
       when(mockUsage.isFirstRun).thenReturn(true);
 
-      memoryFileSystem = MemoryFileSystem();
-      tempDir = memoryFileSystem.systemTempDirectory.createTempSync('flutter_tools_packages_test.');
-      memoryFileSystem.currentDirectory = tempDir;
+      tempDir = fs.systemTempDirectory.createTempSync('flutter_tools_packages_test.');
```

Comment:
Was there a problem with the `MemoryFileSystem` ?

Reply:
It was overriding a parent `tempDir` property. We use `FakeFlutterProjectFactory(tempDir)` in many places to generate a project. It would be good if a project could be generated using the `MemoryFileSystem` without manually adding this boilerplate.